### ISSUE TITLE
[5.8] assertArraySubset deprecated in PHPUnit 8

### ISSUE THREAD

**Issue by GrahamCampbell** [laravel/framework#27550]
Either we need to convince PHPUnit to revert this, or find a replacement, otherwise test code using `Illuminate\Foundation\Testing\TestResponse::assertJson` will start to throw deprecation warnings.

---

Relates to https://github.com/sebastianbergmann/phpunit/issues/3494. // cc @driesvints

**Comment by driesvints** [comment]
No worries. I already took care of this in https://github.com/laravel/framework/pull/27441. I've committed https://github.com/laravel/framework/pull/27441/commits/695a29928d5f3e595363306cf62ba4ff653d73ba which overwrites the method to prevent the warnings. I couldn't take over the full `ArraySubset` class because the method signature on the `Constraint` base class was changed and thus it will only work on either PHPUnit 7 or below or PHPUnit 8 or above. We'll take over the full class as soon as we're dropping support for PHPUnit 7 in February next year.
